safekeeper/src/http/routes.rs

```diff
@@ -373,8 +374,138 @@ async fn dump_debug_handler(mut request: Request<Body>) -> Result<Response<Body>
         .await
         .map_err(ApiError::InternalServerError)?;
 
-    // TODO: use streaming response
-    json_response(StatusCode::OK, resp)
+    use bytes::{Bytes, BytesMut};
+    use std::io::Write as _;
+    use tokio::sync::mpsc;
+    use tokio_stream::wrappers::ReceiverStream;
+    use tracing::info_span;
+
+    /// An [`std::io::Write`] implementation on top of a channel sending [`bytes::Bytes`] chunks.
+    struct ChannelWriter {
+        buffer: BytesMut,
+        tx: mpsc::Sender<std::io::Result<Bytes>>,
+        written: usize,
+    }
+
+    impl ChannelWriter {
+        fn new(buf_len: usize, tx: mpsc::Sender<std::io::Result<Bytes>>) -> Self {
+            assert_ne!(buf_len, 0);
+            ChannelWriter {
+                // split about half off the buffer from the start, because we flush depending on
```

Comment:
this is a copy of code we already have in utils/src/http/endpoint.rs

Reply:
Thanks for comments! @shanyp 

I know it. 
I am working on the PR, step by step.
Atm, I am updating python testing scripts, since this PR introduces streaming instead of json response.
I am adding the parsing logic, similar to `/metrics` response [parsing](https://github.com/neondatabase/neon/blob/main/test_runner/fixtures/neon_fixtures.py#L2938-L2963).

Reply:
thanks for working on this one